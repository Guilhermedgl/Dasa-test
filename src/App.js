import React, { Component } from 'react';
import axios from 'axios';
import Home from './components/home'

class App extends Component {
  constructor() {
    super();
    this.state = {
		  pokemons: [],
    }
    this.getPokemons = this.getPokemons.bind(this);
  }

  async getPokemons() {
    const response = await axios.get('https://pokeapi.co/api/v2/pokemon')
    this.setState({ pokemons: response.data.results });
  }

  async componentDidMount() {
    await this.getPokemons();
  }

  render() {
    return (
      <Home pokemons={this.state.pokemons} />
    );
  }
}

export default App;
