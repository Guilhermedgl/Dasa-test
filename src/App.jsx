import React, { Component } from 'react';
import axios from 'axios';
import Home from './components/home';
import { apiUrl } from './configs/api';

class App extends Component {
  constructor() {
    super();
    this.state = {
      pokemons: [],
    }
    this.getPokemons = this.getPokemons.bind(this);
  }

  componentDidMount() {
    this.getPokemons();
  }

  // GET an array of pokemons from api url
  getPokemons() {
    axios.get(apiUrl)
      .then(response => {
        this.setState({ pokemons: response.data.results });
      })
      .catch(error => console.log(error))
  }


  render() {
    return (
      <Home pokemons={this.state.pokemons} />
    );
  }
}

export default App;
