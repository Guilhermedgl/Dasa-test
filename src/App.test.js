import React from 'react';
import { create } from 'react-test-renderer';
import App from './App';

describe('<App />', () => {

  it('should match with snapshot', () => {
    const component = create(<App />);
    const instance = component.getInstance();
    expect(component.toJSON()).toMatchSnapshot();
    expect(instance.state.pokemons).toHaveLength(0);
  });

  it('should update state with componentDidMount', async () => {
    const component = create(<App />);
    const instance = component.getInstance();
    await instance.componentDidMount();
    expect(instance.state.pokemons.length).toBeGreaterThan(0);
  });
});