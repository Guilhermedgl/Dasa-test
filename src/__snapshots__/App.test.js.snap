// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match with snapshot 1`] = `
<section>
  <nav
    className="navbar"
  >
    <h1>
      Pokemons
    </h1>
  </nav>
  <div
    className="container"
  >
    <div
      className="container-input"
    >
      <input
        className="input"
        onChange={[Function]}
        placeholder="Pesquise por um Pokemon"
        type="text"
      />
    </div>
    <div>
      <h3
        className="notes"
      >
        Notas
      </h3>
    </div>
  </div>
</section>
`;
